_job-scheduler-plugin/build-plugin.md

```diff
@@ -0,0 +1,163 @@
+    public static final String SCHEDULE_FIELD = "schedule";
+    public static final String ENABLED_TIME_FILED = "enabled_time";
+    public static final String ENABLED_TIME_FILED_READABLE = "enabled_time_field";
+    public static final String INDEX_NAME_FIELD = "index_name_to_watch";
+    public static final String LOCK_DURATION_SECONDS = "lock_duration_seconds";
+    public static final String JITTER = "jitter";
+
+    private String jobName;
+    private Instant lastUpdateTime;
+    private Instant enabledTime;
+    private boolean isEnabled;
+    private Schedule schedule;
+    private String indexToWatch;
+    private Long lockDurationSeconds;
+    private Double jitter;
+```
+
+Next, build out your parameters from the declared variables:
+
+```java
+public SampleJobParameter(String id, String name, String indexToWatch, Schedule schedule, Long lockDurationSeconds, Double jitter) {
+        this.jobName = name;
+        this.indexToWatch = indexToWatch;
+        this.schedule = schedule;
+
+        Instant now = Instant.now();
+        this.isEnabled = true;
+        this.enabledTime = now;
+        this.lastUpdateTime = now;
+        this.lockDurationSeconds = lockDurationSeconds;
+        this.jitter = jitter;
+    }
+
+    @Override
+    public String getName() {
+        return this.jobName;
+    }
+
+    @Override
+    public Instant getLastUpdateTime() {
+        return this.lastUpdateTime;
+    }
+
+    @Override
+    public Instant getEnabledTime() {
+        return this.enabledTime;
+    }
+
+    @Override
+    public Schedule getSchedule() {
+        return this.schedule;
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return this.isEnabled;
+    }
+
+    @Override
+    public Long getLockDurationSeconds() {
+        return this.lockDurationSeconds;
+    }
+
+    @Override public Double getJitter() {
+        return jitter;
+    }
+
+    public String getIndexToWatch() {
+        return this.indexToWatch;
+    }
+
+    public void setJobName(String jobName) {
+        this.jobName = jobName;
+    }
+
+    public void setLastUpdateTime(Instant lastUpdateTime) {
+        this.lastUpdateTime = lastUpdateTime;
+    }
+
+    public void setEnabledTime(Instant enabledTime) {
+        this.enabledTime = enabledTime;
+    }
+
+    public void setEnabled(boolean enabled) {
+        isEnabled = enabled;
+    }
+
+    public void setSchedule(Schedule schedule) {
+        this.schedule = schedule;
+    }
+
+    public void setIndexToWatch(String indexToWatch) {
+        this.indexToWatch = indexToWatch;
+    }
+
+    public void setLockDurationSeconds(Long lockDurationSeconds) {
+        this.lockDurationSeconds = lockDurationSeconds;
+    }
+
+    public void setJitter(Double jitter) {
+        this.jitter = jitter;
+    }
+```
+
+The example code above defines the request parameters that can be set when building your plugin. Reference the following table configured by the example code above.
```

Comment:
I think you only need to document the required field in this file https://github.com/opensearch-project/job-scheduler/blob/main/spi/src/main/java/org/opensearch/jobscheduler/spi/ScheduledJobParameter.java